### DISCUSSION TITLE
I want to convert the distance into a result of confidence, any idea?

### DISCUSSION THREAD

**Discussion by coollofty** [milvus-io/milvus#4870] [Q&A and General discussion]
现在的是距离，但距离是不稳定的，如果我想把这个距离转换为[0,100]之间的类似于置信度的这种数值，有什么好的解决思路或方案吗？谢谢

**Comment by yhmo** [comment]
把向量归一化，然后使用MetricType.IP (Inner Product)内积的方式，算出来的distance值是[0, 1]，再映射到[0, 100]的区间里

**Reply by coollofty** [reply]
> 
> 
> 把向量归一化，然后使用MetricType.IP (Inner Product)内积的方式，算出来的distance值是[0, 1]，再映射到[0, 100]的区间里

我使用了以下参数创建集合
cparams = {'collection_name':name, 'dimension':1024, 'index_file_size':1024, 'metric_type':MetricType.IP}

但是search得到的结果是这个样子的：
[[{'id': 30, 'score': 208.32662963867188}, {'id': 30, 'score': 208.32662963867188}, {'id': 30, 'score': 208.32662963867188}]]

score达到了208，并非[0,1]，这是为什么呢？

另外，我用top_k=3搜索，为什么前3个都是一样的呢？也许我的数据有重复，但是却似乎没有办法能list所有的vectors，所以我也不知道是什么问题，像这种情况有没有什么排查方法呢？

谢谢

**Reply by shengjun1985** [reply]
请问你的每条向量都是等长的吗，你之前是用哪种距离计算的？
如果你的向量都是等长的，那么它们的IP或者L2距离都会有确定的最大最小值，这样显然可以映射到[0,100]的区间里。
如果你的向量不是等长的，那么它们可以或者需要被压缩拉长至等长吗（即normalize，归一化）？如果可以的话，那么先归一化，再用IP计算，这样distance值会全部掉在[0, 1]以内。如果不可以，并且一定要用L2计算，那你只能自己定义映射函数了。
通常来讲，有映射到[0,100]区间这种需求的，大都是归一化后的数据。
建议把底库数据和查询向量先归一化看看，看查出来的topk还是不是你想要的东西。
